pkg/cloudprovider/providers/gce/gce_loadbalancer_internal.go

```diff
@@ -128,41 +152,57 @@ func (gce *GCECloud) ensureInternalLoadBalancer(clusterName, clusterID string, s
 	return status, nil
 }
 
-func (gce *GCECloud) updateInternalLoadBalancer(clusterName, clusterID string, service *v1.Service, nodes []*v1.Node) error {
-	igName := getInstanceGroupName(clusterID)
-	_, err := gce.ensureInstanceGroups(igName, nodes)
-	return err
+func (gce *GCECloud) updateInternalLoadBalancer(clusterName, clusterID string, svc *v1.Service, nodes []*v1.Node) error {
+	igName := makeInstanceGroupName(clusterID)
+	igLinks, err := gce.ensureInstanceGroups(igName, nodes)
+	if err != nil {
+		return err
+	}
+
+	// Generate the backend service name
+	_, protocol := getPortsAndProtocol(svc.Spec.Ports)
+	scheme := schemeInternal
+	loadBalancerName := cloudprovider.GetLoadBalancerName(svc)
+	shared := !v1_service.RequestsOnlyLocalTraffic(svc)
+	backendServiceName := makeBackendServiceName(loadBalancerName, clusterID, shared, scheme, protocol)
```

Comment:
Point of using the variable is to show that the scheme must be constant between different resources.